lib/cocoapods/installer/analyzer.rb

```diff
@@ -74,117 +83,111 @@ def analyze(allow_fetches = true)
       #
-      # @return [Bool] Whether the sandbox is in synch with the lockfile.
+      # @return [Array<Source>] the sources to be used in finding
+      #         specifications, as specified by the {#podfile} or all sources.
       #
-      def sandbox_needs_install?(analysis_result)
-        state = analysis_result.sandbox_state
-        needing_install = state.added + state.changed + state.deleted
-        !needing_install.empty?
+      def sources
+        @sources ||= begin
+          sources = podfile.sources
+          plugin_sources = @plugin_sources || []
+
+          # Add any sources specified using the :source flag on individual dependencies.
+          dependency_sources = @podfile_dependency_cache.podfile_dependencies.map(&:podspec_repo).compact
+          all_dependencies_have_sources = dependency_sources.count == @podfile_dependency_cache.podfile_dependencies.count
+
+          if all_dependencies_have_sources
+            sources = dependency_sources
+          elsif has_dependencies? && sources.empty? && plugin_sources.empty?
+            sources = ['https://github.com/CocoaPods/Specs.git']
+          else
+            sources += dependency_sources
+          end
+
+          result = sources.uniq.map do |source_url|
+            config.sources_manager.find_or_create_source_with_url(source_url)
+          end
+          unless plugin_sources.empty?
+            result.insert(0, *plugin_sources)
+          end
+          result
+        end
       end
 
       #-----------------------------------------------------------------------#
 
-      # @!group Configuration
+      private
 
-      # @return [Hash, Boolean, nil] Pods that have been requested to be
-      #         updated or true if all Pods should be updated
-      #
-      attr_accessor :update
+      # @!group Configuration
 
       # @return [Bool] Whether the version of the dependencies which did not
       #         change in the Podfile should be locked.
       #
       def update_mode?
-        update != nil
+        pods_to_update != nil
       end
 
       # @return [Symbol] Whether and how the dependencies in the Podfile
       #                  should be updated.
       #
       def update_mode
-        if !update
+        if !pods_to_update
           :none
-        elsif update == true
+        elsif pods_to_update == true
           :all
-        elsif !update[:pods].nil?
+        elsif !pods_to_update[:pods].nil?
           :selected
         end
       end
 
-      # @return [Bool] Whether the analysis allows pre-downloads and thus
-      #         modifications to the sandbox.
-      #
-      # @note   This flag should not be used in installations.
-      #
-      # @note   This is used by the `pod outdated` command to prevent
-      #         modification of the sandbox in the resolution process.
-      #
-      attr_accessor :allow_pre_downloads
-      alias_method :allow_pre_downloads?, :allow_pre_downloads
```

Comment:
this is always set to `true`. The comment also about `pod outdated` is not true anymore.

Basically this flag is no longer needed.